### PR TITLE
 plank: init at 0.11.4

### PR DESCRIPTION
**Pull request by davidak**
###### Motivation for this change

I want a nice dock.

screenshot_2018-05-08_01-41-33

Also part of the pantheon desktop, that hopefully will be complete some day.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/libraries/bamf/default.nix [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  pname = "bamf";
+  version = "0.5.3";
+  name = "${pname}-${version}";
+
+  src = fetchurl {
+    url = "https://launchpad.net/${pname}/0.5/${version}/+download/${name}.tar.gz";
+    sha256 = "051vib8ndp09ph5bfwkgmzda94varzjafwxf6lqx7z1s8rd7n39l";
+  };
+
+  propagatedBuildInputs = [glib gobjectIntrospection];
```
Comment (Mic92) [inline pkgs/development/libraries/bamf/default.nix]: Why is it propagating here?

File: pkgs/applications/misc/plank/default.nix [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  ];
+
+  buildInputs = [ vala atk cairo glib glibc gnome3.gnome-menus
+                  gtk3 gnome3.libgee libwnck3 libX11 libXfixes
+                  libXi pango gnome3.gnome-common pkgconfig bamf gdk_pixbuf
+                  libdbusmenu-gtk3 gnome3.dconf ];
+
+  propagatedUserEnvPkgs = [
+    gnome3.dconf
+  ];
```
Comment (Mic92) [inline pkgs/applications/misc/plank/default.nix]: Could this be replaced by `wrapGAppsHook`?

File: pkgs/applications/misc/plank/default.nix [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+  nativeBuildInputs = [
+    intltool
+    libxml2 # xmllint
+    wrapGAppsHook
+  ];
+
+  buildInputs = [ vala atk cairo glib glibc gnome3.gnome-menus
+                  gtk3 gnome3.libgee libwnck3 libX11 libXfixes
+                  libXi pango gnome3.gnome-common pkgconfig bamf gdk_pixbuf
```
Comment (Mic92) [inline pkgs/applications/misc/plank/default.nix]: pkgconfig also belongs to nativeBuildInputs.

File: pkgs/applications/misc/plank/default.nix [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    sha256 = "1f41i45xpqhjxql9nl4a1sz30s0j46aqdhbwbvgrawz6himcvdc8";
+  };
+
+  nativeBuildInputs = [
+    intltool
+    libxml2 # xmllint
+    wrapGAppsHook
+  ];
+
+  buildInputs = [ vala atk cairo glib glibc gnome3.gnome-menus
```
Comment (Mic92) [inline pkgs/applications/misc/plank/default.nix]: I don't think you have to add `glibc` here.

File: pkgs/development/libraries/bamf/default.nix (hunk 1 above)
Comment (davidak) [reply pkgs/development/libraries/bamf/default.nix]: i just copied it from some other package. changed

**Comment by davidak** [comment]
@Mic92 thanks for the feedback.
I changed it corresponding to your suggestions.

**Comment by davidak** [comment]
Previously tested with XFCE. Now i can also confirm that it works on GNOME 3 desktop!

**Review by Mic92 — APPROVED** [review]
Can be merged if borg passes the build.

**Pull request merged**